components/constellation/constellation.rs

```diff
@@ -2304,7 +2295,7 @@ impl<Message, LTF, STF> Constellation<Message, LTF, STF>
             }
         }
         for entry in evicted_pipelines {
-            self.close_pipeline(entry.pipeline_id, ExitPipelineMode::Normal);
+            self.close_pipeline(entry.pipeline_id, DiscardBrowsingContext::No, ExitPipelineMode::Normal);
```

Comment:
This code is used when a browsing context navigates, we don't want to close the BC in this case. Nested BCs will get closed, when the recursive call is made it's with `DiscardBrowsingContext::Yes`.